server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -333,20 +338,20 @@ public Tablet(final TabletServer tabletServer, final KeyExtent extent,
     boolean replicationEnabled =
         ReplicationConfigurationUtil.isEnabled(extent, this.tableConfiguration);
     TabletFiles tabletPaths =
-        new TabletFiles(data.getDirectory(), data.getLogEntries(), data.getDataFiles());
+        new TabletFiles(data.getDirectoryName(), data.getLogEntries(), data.getDataFiles());
     tabletPaths = VolumeUtil.updateTabletVolumes(tabletServer.getContext(), tabletServer.getLock(),
         fs, extent, tabletPaths, replicationEnabled);
 
-    // deal with relative path for the directory
-    Path locationPath;
-    if (tabletPaths.dir.contains(":")) {
-      locationPath = new Path(tabletPaths.dir);
-    } else {
-      locationPath = tabletServer.getFileSystem().getFullPath(FileType.TABLE,
-          extent.getTableId() + tabletPaths.dir);
-    }
-    this.location = locationPath;
-    this.tabletDirectory = tabletPaths.dir;
+    // TODO may be needed in upgrade
+    /*
+     * // deal with relative path for the directory Path locationPath; if
+     * (tabletPaths.dir.contains(":")) { locationPath = new Path(tabletPaths.dir); } else {
+     * locationPath = tabletServer.getFileSystem().getFullPath(FileType.TABLE, extent.getTableId() +
+     * tabletPaths.dir); }
+     */
+
```

Comment:
LGTM.  Just don't forget to make this TODO a ticket.  I have found TODOs are likely to get lost unless they have a ticket.

Reply:
I addressed all todos